akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/DynamoDBOffsetStore.scala

```diff
@@ -45,25 +45,10 @@ private[projection] object DynamoDBOffsetStore {
   type SeqNr = Long
   type Pid = String
 
-  final case class Record(slice: Int, pid: Pid, seqNr: SeqNr, timestamp: Instant) extends Ordered[Record] {
-
-    override def compare(that: Record): Int = {
-      val result = this.timestamp.compareTo(that.timestamp)
-      if (result == 0) {
-        if (this.slice == that.slice)
-          if (this.pid == that.pid)
-            if (this.seqNr == that.seqNr)
-              0
-            else
-              java.lang.Long.compare(this.seqNr, that.seqNr)
-          else
-            this.pid.compareTo(that.pid)
-        else Integer.compare(this.slice, that.slice)
-      } else {
-        result
-      }
-    }
-  }
+  final case class Record(slice: Int, pid: Pid, seqNr: SeqNr, timestamp: Instant)
+
+  // Used only in eviction, does not imply anything about equals
+  private val recordOrdering: Ordering[Record] = Ordering.by(_.timestamp)
```

Comment:
`Ordered` carries with it object equality connotations (that `a.compare(b) == 0` implies `a.equals(b)`) that we don't really need.